packages/components/pie-button/src/button.scss

```diff
@@ -8,29 +8,33 @@
 
 /**
 * Defines button interactive states for hover, active and loading.
-* Takes the interactive color name as an argument, and whether the color
-* should be lightened or darkened.
+* $bg-color: design token name to be used for the states
+* $mode: 'default': apply the default lighten/darken to the color via hsl
+*        'transparent': uses hsla syntax to make the color semi-opaque (used for ghost and outline button variants)
 */
-@mixin button-interactive-states($bg-color, $lighten: false) {
+@mixin button-interactive-states($bg-color, $mode: 'default') {
     &:hover:not(:disabled) {
         --hover-modifier: calc(-1 * var(--dt-color-hover-01));
 
-        @if $lighten {
-            --hover-modifier: var(--dt-color-hover-02);
+        // for mode=transparent, use the hsla syntax to make the button background opaque by a set percentage
+        @if $mode == 'transparent' {
+            --hover-modifier: var(--dt-color-hover-01);
+            --btn-bg-color: hsl(var(#{$bg-color}-h), var(#{$bg-color}-s), var(#{$bg-color}-l), var(--hover-modifier));
```

Comment:
So for black, does this resolve to `hsl(0, 0%, 0%, -4%)`?

Reply:
Close – `hsl(0, 0%, 0%, 4%)`